### ISSUE TITLE
Extra tags in multiline regex tables

### ISSUE THREAD

**Issue by bernardjosef** [universal-ctags/ctags#2795]
Multiline regex tables do not support conditional tagging with extra. The following patch seems to solve the problem, but I do not know if there are drawbacks.

````
diff -urN ctags-orig/main/lregex.c ctags/main/lregex.c
--- ctags-orig/main/lregex.c	2021-01-06 18:31:30.000000000 +0100
+++ ctags/main/lregex.c	2021-01-08 02:45:40.000000000 +0100
@@ -2407,6 +2407,11 @@
 	{
 		regexTableEntry *entry = ptrArrayItem(table->entries, i);
 		regexPattern *ptrn = entry->pattern;
+
+		if ((entry->pattern->xtagType != XTAG_UNKNOWN)
+			&& (!isXtagEnabled (entry->pattern->xtagType)))
+			continue;
+
 		struct guestSpec  *guest = &ptrn->guest;
 
 		Assert (ptrn);
````

**Comment by masatake** [comment]
Thank you very much. You are correct. Can I ask you to make a pull request for this?
You can move the lines upward like:
```
 		regexTableEntry *entry = ptrArrayItem(table->entries, i);
+		if ((entry->pattern->xtagType != XTAG_UNKNOWN)
+			&& (!isXtagEnabled (entry->pattern->xtagType)))
+			continue;
+
 		regexPattern *ptrn = entry->pattern;
 		struct guestSpec  *guest = &ptrn->guest;
```

Could you add a test case I wrote to the pull request?
For adding, do "git am 0001-Tmain-add-a-case-for-testing-_extra-mtable-regex-fla.txt"
[0001-Tmain-add-a-case-for-testing-_extra-mtable-regex-fla.txt](https://github.com/universal-ctags/ctags/files/5790000/0001-Tmain-add-a-case-for-testing-_extra-mtable-regex-fla.txt)
.

**Comment by masatake** [comment]
I wonder how do you find this bug because `_extra` flag is so minor.
You will be interested in my blog article. https://ctags.io/2021/01/05/optscript/.

**Comment by bernardjosef** [comment]
I am currently working on a parser for Markdown wiki links and Pandoc citations in Markdown files for a Zettelkasten (slip box). This is why I have found the bug, because I have defined an extra flag there. My goal is rather unusual for Universal ctags, I believe. I want to follow links and citations inside a Zettelkasten with the help of tags.

**Comment by masatake** [comment]
> I am currently working on a parser for Markdown wiki links ...

Are you extending optlib/markdown.ctags or writing a new markdown parser from scratch?

I have tried to tag links in a markdown file. However, I found it is not so easy.
```markdown
    [link-I-would-like-to-tag] (URL)
```

Though the markdown parser in u-ctags is written in `_mtable-regex-flags`, it is line oriented. 
I think full-rewriting may be needed to support the links.

If you feel the current version of u-ctags is not enough for your goal, rememer #2794. I'm implementing a general purpose interpreter to u-ctags command line.
